mesonbuild/utils/universal.py

```diff
@@ -1361,34 +1365,43 @@ def do_conf_file(src: str, dst: str, confdata: 'ConfigurationData',
 
 '''
 
-def dump_conf_header(ofilename: str, cdata: 'ConfigurationData', output_format: T.Literal['c', 'nasm']) -> None:
-    if output_format == 'c':
-        prelude = CONF_C_PRELUDE
-        prefix = '#'
-    else:
+def _dump_c_header(ofile: TextIOWrapper, output_format: str, cdata: ConfigurationData) -> None:
+    format_desc: T.Callable[[str], str]
+    if output_format == 'nasm':
         prelude = CONF_NASM_PRELUDE
         prefix = '%'
+        format_desc = lambda desc: '; ' + '\n; '.join(desc.splitlines()) + '\n'
+    else:  # c
+        prelude = CONF_C_PRELUDE
+        prefix = '#'
+        format_desc = lambda desc: '/* ' + '\n'.join(desc.splitlines()) + ' */\n'
```

Comment:
What's the purpose of the change from:
```
'%s */\n' % desc
```
to
```
'\n'.join(desc.splitlines())
```

We're splitting lines then joining them again as lines?

Reply:
oups. That's what happen when you refactor a refactor...